src/Data/Json/Extended/Signature/Core.purs

```diff
@@ -127,9 +129,59 @@ intToHugeNum =
   HN.fromNumber
     <<< Int.toNumber
 
-derive instance ordEJsonF ∷ Ord a ⇒ Ord (EJsonF a)
+instance ordEJsonF ∷ Ord a ⇒ Ord (EJsonF a) where
+  compare = compare1
+
+-- Note: this cannot be derived, due to integer/decimal comparisons and map
+-- comparisons
 instance ord1EJsonF ∷ Ord1 EJsonF where
-  compare1 = compare
+  compare1 Null Null = EQ
+  compare1 _ Null = GT
+  compare1 Null _ = LT
+
+  compare1 (Boolean b1) (Boolean b2) = compare b1 b2
+  compare1 _ (Boolean _) = GT
+  compare1 (Boolean _) _ = LT
+
+  compare1 (Integer i) (Integer j) = compare i j
+  compare1 (Integer i) (Decimal b) = compare (intToHugeNum i) b
```

Comment:
Okidoke. We'll need to change `Eq` correspondingly then too - it also has a case for equality of integers and decimals.